### ISSUE TITLE
Strange behavior with ul/ol tags in RichTextLabel

### ISSUE THREAD

**Issue by Rindbee** [godotengine/godot#61621]
### Godot version

v4.0.alpha6.official [e4f0fc50f] v4.0.alpha.custom_build [6cd730ea9]

### System information

Linux Mint 20.3, Vulkan, NVIDIA GeForce GTX 1050 Ti

### Issue description

1. Strange behavior near ul/ol close tags;
2. Strange behavior with **empty** ul/ol tags;
3. Strange behavior with nested ul/ol tags.

There are 3 points, and they might be essentially the same one. 
![1](https://user-images.githubusercontent.com/30386067/171555857-69e09ef4-d68d-403b-8ce7-7aa9d3418e7a.png) ![captrue](https://user-images.githubusercontent.com/30386067/171555861-cfabaec7-bb4e-4cf2-9e80-697bf5cecf13.png)

### Steps to reproduce

1. Add a `RichTextLabel`
2. Check **Bbcode Enable**
3. In the Text field, enter the following text.
```
[ol]
two
three[ul]a
b
c[/ul]four
[/ol]end
test 1
a
[ul][/ul]b
c
a
[ol][/ol]b
c
```

**Comment by Calinou** [comment]
Do these issues occur on 3.4.4/3.5rc2?

**Comment by kleonc** [comment]
> Do these issues occur on 3.4.4/3.5rc2?

No, as list tags (`ul`, `ol`) are not supported in `3.x`, see #49914.

**Comment by YuriSizov** [comment]
For the reference, the first two points were resolved by the linked PR.
